tensorflow/core/ops/training_ops.cc

```diff
@@ -1053,4 +1053,94 @@ REGISTER_OP("ResourceApplyPowerSign")
       return ApplyPowerSignShapeFn(c, /*sparse=*/false);
     });
 
+static Status ApplyRpropShapeFn(InferenceContext* c, bool withError) {
```

Comment:
(For tf-api-owners)

The op and kernel should also live in tensorflow/contrib instead of in tensorflow/core?